src/utils/Options.ts

```diff
@@ -18,84 +37,131 @@ module n3Charts.Utils {
       }
 
-      return this.axes[side];
+      var series = [];
+
+      // Extend the default series options
+      angular.extend(series, Options.DEFAULT.series, jsSeries);
+
+      return this.sanitizeSeries(series);
     }
 
-    getAbsKey(): string {
-      if (!this.axes[Factory.Axis.SIDE_X]) {
-        throw new TypeError('Cannot find abs key : ' + Factory.Axis.SIDE_X);
+    parseAxes(jsAxes: any): IAxes {
+      // Type check because of <any> type
+      if (!angular.isObject(jsAxes)) {
+        throw TypeError('Axes option must be an object.');
       }
 
-      return this.axes[Factory.Axis.SIDE_X].key;
+      var axes = {};
+
+      // Extend the default axes options
+      angular.extend(axes, Options.DEFAULT.axes, jsAxes);
+
+      return this.sanitizeAxes(axes);
     }
 
     sanitizeSeries(series: any[]) {
-      return (series || []).map((s) => new Series(s));
+      return (series).map((s) => new SeriesOptions(s));
     }
 
     sanitizeAxes(axes: any) {
-      // Map operation over an object, that returns a new object
-      return Object.keys(axes || {}).reduce((prev, key) => {
+      // Map object keys and return a new object
+      return <IAxes> Object.keys(axes).reduce((prev, key) => {
         prev[key] = new AxisOptions(axes[key]);
         return prev;
       }, {});
     }
 
-    _isValidSeriesType(type:string): Boolean {
-      for (var key in Options.SERIES_TYPES) {
-        if (Options.SERIES_TYPES[key] === type) {
-          return true;
-        }
+    sanitizeMargin(margin) {
+      return <IMargin> {
+        top: parseFloat(margin.top),
+        left: parseFloat(margin.left),
+        bottom: parseFloat(margin.bottom),
+        right: parseFloat(margin.right)
+      };
+    }
+
+    isValidAxisSide(side:string): Boolean {
+      return d3.values(Factory.Axis.SIDE).indexOf(side) === -1
+        ? false : true;
+    }
+
+    isValidSeriesType(type:string): Boolean {
+      return d3.values(Options.SERIES_TYPES).indexOf(type) === -1
+        ? false : true;
+    }
+
+    getAbsKey(): string {
+      if (!this.axes[Factory.Axis.SIDE.X]) {
+        throw new TypeError('Cannot find abs key : ' + Factory.Axis.SIDE.X);
       }
 
-      return false;
+      return this.axes[Factory.Axis.SIDE.X].key;
+    }
+
+    getByAxisSide(side: string) {
+        if (!this.isValidAxisSide(side)) {
+            throw new TypeError('Cannot get axis side : ' + side);
+        }
+
+        return this.axes[side];
     }
 
-    getSeriesByType(type: string): Series[] {
-      if (this._isValidSeriesType(type) === false) {
+    getSeriesByType(type: string): SeriesOptions[] {
+      if (!this.isValidSeriesType(type)) {
         throw new TypeError('Unknown series type: ' + type);
       }
 
-      return this.series.filter((s) =>
-        s.type.indexOf(type) > -1
-      );
+      return this.series.filter((series) => series.type.indexOf(type) !== -1);
+    }
+
+    public static uuid(): string {
```

Comment:
I added this function to create an ID for series, where no ID is defined. Not sure if this is a good idea, but I think it's nicer than to throw an error if no ID is defined.